### PR TITLE
Bump qr-code-generator/1.7.0

### PR DESCRIPTION
**Pull request by Nekto89**
- remove languages from description
- modernize recipe
- update test_package because of changed header name
- integrate patch into CMakeLists.txt

Specify library name and version:  **qr-code-generator/1.7.0**

I've noticed that there is a new version available so I've decided to bump it in CCI.
Version check is based on existing files. It's not pretty but works.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/qr-code-generator/all/CMakeLists.txt [hunk 1]
```diff
@@ -24,19 +24,28 @@ if(WIN32 AND BUILD_SHARED_LIBS)
     set(CMAKE_VS_INCLUDE_INSTALL_TO_DEFAULT_BUILD TRUE)
 endif()
 
-set(SRC
-    source_subfolder/cpp/QrCode.cpp
-)
+if(EXISTS ${CMAKE_CURRENT_LIST_DIR}/source_subfolder/cpp/QrSegment.cpp)
```
Comment (madebr) [inline recipes/qr-code-generator/all/CMakeLists.txt:27]: Won't it make more sense to move this to patches?
e.g. let each version create a `files.cmake` file that defines these variables and include that here.

File: recipes/qr-code-generator/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -4,7 +4,11 @@ project(test_package)
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
 conan_basic_setup()
 
+find_package(qr-code-generator REQUIRED)
 add_executable(${PROJECT_NAME} test_package.cpp)
-target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
+target_link_libraries(${PROJECT_NAME} qr-code-generator::qr-code-generator)
+if (qr-code-generator_VERSION VERSION_LESS "1.7.0")
```
Comment (madebr) [inline recipes/qr-code-generator/all/test_package/CMakeLists.txt:10]: I guess you need to pass this flag as an argument to cmake in the test package's conanfile.py.

File: recipes/qr-code-generator/all/patches/002-cmake-pre-1.6.patch [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+ 
+-set(SRC source_subfolder/cpp/qrcodegen.cpp)
+-set(HEADERS source_subfolder/cpp/qrcodegen.hpp)
+-set(LIBRARY_NAME qrcodegencpp)
++set(SRC source_subfolder/cpp/QrCode.cpp source_subfolder/cpp/QrSegment.cpp source_subfolder/cpp/BitBuffer.cpp)
++set(HEADERS source_subfolder/cpp/QrCode.hpp source_subfolder/cpp/QrSegment.hpp source_subfolder/cpp/BitBuffer.hpp)
++set(LIBRARY_NAME qrcodegen)
+ 
+ add_library(${LIBRARY_NAME} ${SRC})
+
```
Comment (madebr) [inline recipes/qr-code-generator/all/patches/002-cmake-pre-1.6.patch:15]: I meant a patch like this:
```suggestion
--- /dev/null
+++ files.cmake
@@ -0,0 +1,3 @@
+set(SRC source_subfolder/cpp/QrCode.cpp source_subfolder/cpp/QrSegment.cpp source_subfolder/cpp/BitBuffer.cpp)
+set(HEADERS source_subfolder/cpp/QrCode.hpp source_subfolder/cpp/QrSegment.hpp source_subfolder/cpp/BitBuffer.hpp)
+set(LIBRARY_NAME qrcodegen)
```

File: recipes/qr-code-generator/all/CMakeLists.txt [hunk 4]
```diff
@@ -24,19 +24,15 @@ if(WIN32 AND BUILD_SHARED_LIBS)
     set(CMAKE_VS_INCLUDE_INSTALL_TO_DEFAULT_BUILD TRUE)
 endif()
 
-set(SRC
-    source_subfolder/cpp/QrCode.cpp
-)
+set(SRC source_subfolder/cpp/qrcodegen.cpp)
+set(HEADERS source_subfolder/cpp/qrcodegen.hpp)
+set(LIBRARY_NAME qrcodegencpp)
```
Comment (madebr) [inline recipes/qr-code-generator/all/CMakeLists.txt:29]:
```suggestion
include("${CMAKE_CURRENT_LIST_DIR}/files.cmake")
```

**Comment by Nekto89** [comment]
> @Nekto89 hello, I also kindly ask you to see [our repo](https://github.com/EasyCoding/qrcodegen-cmake) with CMake config (proposed for upstream [here](https://github.com/nayuki/QR-Code-generator/pull/72#issuecomment-895277657) ) and integrate some parts from it :)
> 

I wasn't planning on making larger changes. This requires adding new options, introducing components, adding test for C interface, etc. IMO it should be done in separate PR.

**Review by SSE4 — APPROVED** [review]

**Comment by SSE4** [comment]
> I wasn't planning on making larger changes. This requires adding new options, introducing components, adding test for C interface, etc. IMO it should be done in separate PR.

agree, no need to do everything at one shot

**Review by madebr — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**